brave/browser-laptop
# Using Chromium's spellchecker

Test Plan:
--
a. Suggestions

1. Go to `about:styles` 
2. Type `braave` in textbox
3. There should be redline under `braave`
4. Open context menu on `braave`
5. You can see `bravo`, `brave`, `Brava`
6. Click on `brave`
7. The text should be replaced with `brave` with no redline

b. Learn Spelling
1. Go to `about:styles` 
2. Type `braave` in textbox
3. There should be redline under `braave`
4. Open context menu on `braave`
5. Click `Learn Spelling`
6. `braave` will no longer be redlined 
7.  Next time you type `braave`, you will not see redline

c. Forget Learned Spelling
1. Go to `about:styles` 
2. Type `braave` in textbox
3. There should be redline under `braave`
4. Open context menu on `braave`
5. Click `Learn Spelling`
6. `braave` will no longer be redlined 
7. Open context menu on `braave`
8. Click `Forget Learned Spelling`
9.  Next time you type `braave`, you will see redline under it

d. Legacy data migration
1. Use older Brave to `Learn Spelling` for `braave` and `Ignore Spelling` for `braaave`
2. Update to the version with chromium spellchecker
3. There will be no `dictionary` structure in session store
4. You can see those two words will be added to `Custom Dictionary.txt` under user folder
5. And try to type `braave` and `braaave` in `about:styles`
6. They should be valid words
---
Auditors: @bridiver, @bbondy, @bsclifton 

fixes #9880 

requires https://github.com/brave/muon/pull/244

Submitter Checklist:


Reviewer Checklist:

Tests
